corehq/apps/app_manager/util.py

```diff
@@ -346,20 +346,24 @@ def get_session_schema(form):
     """Get form session schema definition
     """
     structure = {}
-    # TODO handle advanced modules with more than one case
-    if hasattr(form, 'get_module'):
-        case_type = form.get_module().case_type
-    else:
-        case_type = None
-
-    if case_type:
-        structure["case_id"] = {
+    cases = []
+    # TODO handle advanced modules with more than one use of the same case
+    if form.doc_type == "AdvancedForm":
+        for action in form.actions.load_update_cases:
+            cases.append((action.case_type, action.case_session_var))
+    elif hasattr(form, 'get_module'):
+        if form.get_module().case_type:
+            cases.append((form.get_module().case_type, 'case_id'))
```

Comment:
These session vars can get changed at app build time depending on the workflow. This is how we do it for cloudcare: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/touchforms_api.py#L118-L124